code/modules/admin/verbs/debug.dm

```diff
@@ -70,17 +70,38 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
 			return
 		log_admin("[key_name(src)] called [target]'s [procname]() with [lst.len ? "the arguments [list2params(lst)]":"no arguments"].")
 		message_admins("[key_name(src)] called [target]'s [procname]() with [lst.len ? "the arguments [list2params(lst)]":"no arguments"].")
-		returnval = call(target,procname)(arglist(lst)) // Pass the lst as an argument list to the proc
+		returnval = WrapAdminProcCall(target, procname, lst) // Pass the lst as an argument list to the proc
 	else
 		//this currently has no hascall protection. wasn't able to get it working.
 		log_admin("[key_name(src)] called [procname]() with [lst.len ? "the arguments [list2params(lst)]":"no arguments"].")
 		message_admins("[key_name(src)] called [procname]() with [lst.len ? "the arguments [list2params(lst)]":"no arguments"].")
-		returnval = call(procname)(arglist(lst)) // Pass the lst as an argument list to the proc
+		returnval = WrapAdminProcCall(GLOBAL_PROC, procname, lst) // Pass the lst as an argument list to the proc
 	. = get_callproc_returnval(returnval, procname)
 	if(.)
 		to_chat(usr, .)
 	feedback_add_details("admin_verb","Advanced ProcCall") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
 
+GLOBAL_VAR_INIT(AdminProcCall, null)
+GLOBAL_PROTECT(AdminProcCall)
+
+/proc/WrapAdminProcCall(target, procname, list/arguments)
+	if(GLOB.AdminProcCall)
+		to_chat(usr, "<span class='adminnotice'>Another admin called proc is still running, your proc will be run after theirs finishes</span>")
+		UNTIL(!GLOB.AdminProcCall)
+		to_chat(usr, "<span class='adminnotice'>Running your proc</span>")
+	GLOB.AdminProcCall = usr.client.ckey	//if this runtimes, too bad for you
+	try
+		if(target == GLOBAL_PROC)
+			. = call(procname)(arglist(arguments))
+		else
+			. = call(procname)(arglist(arguments))
+	catch()
```

Comment:
The issue with using a try catch, is that while runtimes normally only unwind the current proc, a runtime in a stack with a try/catch will unwind the entire stack, rather than just continuing.

Its minor, but it means behavior is changed from admin proc calls compared to normal proc calls, and there are things that assume a runtime in a proc they called won't stop them, so state could get fucked up.